contracts/InvestStrategyClient.sol

```diff
@@ -0,0 +1,113 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.0;
+
+import {Address} from "@openzeppelin/contracts/utils/Address.sol";
+import {IERC20Metadata} from "@openzeppelin/contracts/token/ERC20/extensions/IERC20Metadata.sol";
+import {IInvestStrategy} from "./interfaces/IInvestStrategy.sol";
+
+/**
+ * @title InvestStrategyClient
+ * @dev Library to simplify the interaction with IInvestStrategy objects
+ * @custom:security-contact security@ensuro.co
+ * @author Ensuro
+ */
+library InvestStrategyClient {
+  using Address for address;
+
+  event StrategyChanged(IInvestStrategy oldStrategy, IInvestStrategy newStrategy);
+  event WithdrawFailed(bytes reason);
+  event DepositFailed(bytes reason);
+  event DisconnectFailed(bytes reason);
+
+  function dcConnect(IInvestStrategy strategy, bytes memory initStrategyData) internal {
+    address(strategy).functionDelegateCall(abi.encodeWithSelector(IInvestStrategy.connect.selector, initStrategyData));
```

Comment:
Better to use abi.encodeCall which does type checking: https://github.com/OpenZeppelin/openzeppelin-contracts/issues/3693